### DISCUSSION TITLE
Search by ID when using UUID or ULID

### DISCUSSION THREAD

**Discussion by gabrielgasp** [spatie/laravel-permission#2373] [Ideas]
Currently, when performing operations such as `assignRole`, `removeRole`, `givePermissionTo`, etc., the default behavior is to search for roles and permissions using either their numeric ID or their name if the argument provided is a string. However, this can pose a problem if you're using UUIDs or ULIDs as primary keys, since they are also strings but should still be treated as IDs.

To address this issue, I suggest implementing a check using the `Str::isUuid` and `Str::isUlid` methods before searching by name. This will ensure that UUIDs and ULIDs are properly recognized as IDs and searched for accordingly.

While this issue may not affect most users, would be nice to address it to ensure the consistency of the package's behavior.

**Comment by erikn69** [comment]
There is a V6 new feature, but it still has not been released

- https://github.com/spatie/laravel-permission/pull/2089

Maybe you could test it

**Reply by gabrielgasp** [reply]
I'm pleased to hear that the issue is already being taken care of. I've tested the changes in the pull request on my local environment, and I'm happy to report that they worked flawlessly.

I appreciate your hard work and commitment to this project. Thank you!
